src/Databases/DatabasesCommon.cpp

```diff
@@ -149,6 +149,12 @@ ASTPtr getCreateQueryFromStorage(const StoragePtr & storage, const ASTPtr & ast_
                         return nullptr;
                 }
                 ast_column_declaration->type = ast_type;
+
+                if (auto column_default = metadata_ptr->columns.getDefault(column_name_and_type.name))
+                {
+                    ast_column_declaration->default_specifier = toString(column_default->kind);
+                    ast_column_declaration->default_expression = column_default->expression;
```

Comment:
What about `ast_column_declaration->ttl` and `ast_column_declaration->collation`?

Reply:
JFYI The `getCreateQueryFromStorage` was initially added to make SHOW CREATE work for system tables that don't have an .sql file (system tables don't have default expressions, codecs, and ttl, so it was missing)

Reply:
But it would be nice to avoid similar bugs in the future, since we started using this method for another purpose. For example, we can add an assertion, that will compare the result of `getCreateQueryFromStorage` with the actual CREATE query when creating a table (so tests will check that nothing is missing in `getCreateQueryFromStorage`)